### PR TITLE
Return `Time` when using `time` properties

### PR DESCRIPTION
**Pull request by vhirtham**
## Changes

Return instances of `Time` when using the `time` property of the LCS or `TimeSeries`

## Related Issues

Closes # (add issue numbers)

## Checks

- [x] updated CHANGELOG.md
- [x] updated tests
- [x] updated doc/
- [x] update example/tutorial notebooks

### CODE REVIEW

File: tutorials/welding_example_02_weaving.ipynb [hunk 1]
```diff
@@ -531,7 +531,9 @@
    "outputs": [],
    "source": [
     "t_interp = pd.timedelta_range(\n",
-    "    start=tcp_wire.time[0].values, end=tcp_wire.time[-1].values, freq=\"20ms\"\n",
+    "    start=tcp_wire.time.as_pandas_index()[0], \n",
+    "    end=tcp_wire.time.as_pandas_index()[-1], \n",
+    "    freq=\"20ms\"\n",
```
Comment (CagtayFabry) [inline tutorials/welding_example_02_weaving.ipynb]: can you try `tcp_wire.time[0]` here now with `__getitem__` @vhirtham ?

Comment (vhirtham) [reply tutorials/welding_example_02_weaving.ipynb]: See commit 5dcd477888d035797bda162586763ff758876166 😉

Comment (CagtayFabry) [reply tutorials/welding_example_02_weaving.ipynb]: nice :)

**Comment by vhirtham** [comment]
Did some optimizations. The `Time` class is really nice 😄

**Review by CagtayFabry — APPROVED** [review]

**Pull request merged**